is_core/generic_views/form_views.py

```diff
@@ -576,37 +585,43 @@ def pre_save_obj(self, obj, form, change):
     def post_save_obj(self, obj, form, change):
         self.core.post_save_model(self.request, obj, form, change)
 
-    def get_exclude(self):
-        return (self.exclude is not None and self.exclude or
-                self.core.get_ui_form_exclude(self.request, self.get_obj(True)))
-
     def get_readonly_fields(self):
-        return (self.readonly_fields is not None and self.readonly_fields or
-                self.core.get_form_readonly_fields(self.request, self.get_obj(True)))
+        if self.readonly_fields is not None:
+            return self.readonly_fields
+
+        readonly_fields = self.core.get_readonly_fields(self.request, self.get_obj(True))
+        return () if readonly_fields is None else readonly_fields
 
     def get_fieldsets(self):
-        return (self.fieldsets is not None and self.fieldsets or
-                self.core.get_form_fieldsets(self.request, self.get_obj(True)))
+        if self.fieldsets is not None:
+            return self.fieldsets
+
+        return self.core.get_fieldsets(self.request, self.get_obj(True))
 
     def get_fields(self):
-        return (self.fields is not None and self.fields or
-                self.core.get_ui_form_fields(self.request, self.get_obj(True)))
+        if self.fields is not None:
+            return self.fields
+
+        fields = self.core.get_fields(self.request, self.get_obj(True))
+        return () if fields is None else fields
```

Comment:
What about this?
```python
return fields or ()
```

Reply:
this is not the same 

fields is None != bool(fields)

Reply:
I know but it depends what you expect in `fields`. Your solution is fine if you need detect exactly `None`.